src/app/components/mobile/mobile.component.ts

```diff
@@ -0,0 +1,93 @@
+import { Component, Input, OnChanges, OnDestroy, OnInit, SimpleChanges } from '@angular/core';
+import { BehaviorSubject, Subscription } from 'rxjs';
+import { chatData } from 'src/app/models/chat-data.model';
+import { ChatService } from 'src/app/services/chat.service';
+
+@Component({
+  selector: 'app-mobile',
+  templateUrl: './mobile.component.html',
+  styleUrls: ['./mobile.component.css']
+})
+
+export class MobileComponent implements OnInit, OnDestroy {
+  public textInput = '';
+  public colourList = [ {colorCode: '#ffffff', isSelected: false}, 
+                        {colorCode: '#dce2c6', isSelected: false},
+                        {colorCode: '#677879', isSelected: false},
+                        {colorCode: '#284017', isSelected: false},
+                        {colorCode: '#4e4e6c', isSelected: false},
+                        {colorCode: '#5d82a0', isSelected: false},
+                      ];
+  @Input() public senderName !: string;
+  @Input() public receiverName !: string;
+  @Input() public colourSelected !:string;
+  public chatData!:chatData;
+  public chatSubscription!: Subscription;
+  
+  constructor(private chatService: ChatService) {
+  }
+  
+  ngOnInit():void {
+    let index = this.colourList.findIndex((color) => color.colorCode === localStorage.getItem(this.senderName));
+    this.colourList[index].isSelected = true;
+
+    this.chatSubscription = this.chatService.chat.subscribe(
```

Comment:
Subscription is good but we can simply use child & parent communication for this use case and avoid as much subs as possible